lib/segment/src/index/sparse_index/sparse_vector_index.rs

```diff
@@ -584,19 +584,28 @@ impl<TInvertedIndex: InvertedIndex> VectorIndex for SparseVectorIndex<TInvertedI
         self.inverted_index.vector_count()
     }
 
-    fn update_vector(&mut self, id: PointOffsetType, vector: VectorRef) -> OperationResult<()> {
+    fn update_vector(
+        &mut self,
+        id: PointOffsetType,
+        vector: VectorRef,
+        old_vector: Option<Vector>,
+    ) -> OperationResult<()> {
         if self.config.index_type != SparseIndexType::MutableRam {
             return Err(OperationError::service_error(
                 "Cannot update vector in non-appendable index",
             ));
         }
 
         let vector: &SparseVector = vector.try_into()?;
+        let old_vector: Option<SparseVector> =
+            old_vector.map(SparseVector::try_from).transpose()?;
+
         // do not upsert empty vectors into the index
         if !vector.is_empty() {
```

Comment:
We need to delete the old vector even the new vector is empty.

Reply:
Good catch!